### PR TITLE
Pin PostgreSQL using build-variants

### PR DESCRIPTION
**Pull request by sodre**

### CODE REVIEW

**Comment by sodre** [comment]
@conda-forge-admin, please rerender

File: ci_support/run_docker_build.sh [hunk 1]
```diff
@@ -58,7 +58,7 @@ conda clean --lock
 conda install --yes --quiet conda-forge-build-setup
 source run_conda_forge_build_setup
 
-conda build /recipe_root --quiet || exit 1
+conda build -m /opt/conda/conda_build_config.yaml /recipe_root --quiet || exit 1
```
Comment (sodre) [inline ci_support/run_docker_build.sh]: This line needs to be removed before merging!

**Comment by isuruf** [comment]
@conda-forge-admin, please rerender

File: recipe/meta.yaml [hunk 2]
```diff
@@ -2,11 +2,9 @@
 {% set version = "2.4.3" %}
 {% set sha256 = "ea5374c5db6b645ba5628ddcb08f71d3b3d90a464d366b4e1d20d5a268bde4b9" %}
 
-{% set postgresql_version = "10.3" %}
-
 package:
   name: {{ name|lower }}
-  version: {{ version }}+{{ postgresql_version }}
+  version: {{ version }}
```
Comment (isuruf) [inline recipe/meta.yaml]: You can keep this with `{{ version }}+{{ postgresql }}`

File: recipe/meta.yaml [hunk 3]
```diff
@@ -22,23 +20,23 @@ build:
...
+    - geos
     - gettext
     - pcre
-    - proj4 4.9.3
-    - json-c 0.12|0.12.*
+    - proj4
+    - json-c 
     - perl
-    - postgresql {{ postgresql_version }}
+    - postgresql {{ postgresql }}
```
Comment (isuruf) [inline recipe/meta.yaml]: You can change this to just `- postgresql`

File: recipe/meta.yaml (hunk 2 above)
Comment (isuruf) [reply recipe/meta.yaml]: It doesn't matter. Your choice

File: recipe/meta.yaml [hunk 4]
```diff
@@ -22,23 +20,23 @@ build:
...
-    - geos 3.6.2
+    - gdal
+    - geos
     - gettext
     - pcre
-    - proj4 4.9.3
-    - json-c 0.12|0.12.*
-    - postgresql {{ postgresql_version }}
+    - proj4
+    - {{ pin_compatible('json-c') }}
```
[the remark this answers is not in the document]
Comment (sodre) [reply recipe/meta.yaml]: with `conda render`

[the remark this answers is not in the document]
Comment (sodre) [reply recipe/meta.yaml]: ignore what I said. looks like I had some other problem...

[the remark this answers is not in the document]
Comment (isuruf) [reply recipe/meta.yaml]: You should give the config file to `conda render` with `-m`

File: recipe/meta.yaml [hunk 5]
```diff
@@ -22,23 +20,23 @@ build:
...
     - gettext
     - pcre
-    - proj4 4.9.3
-    - json-c 0.12|0.12.*
+    - proj4
+    - json-c 
     - perl
-    - postgresql {{ postgresql_version }}
+    - postgresql
     - toolchain
```
Comment (isuruf) [inline recipe/meta.yaml]: Change this to `{{ compiler('c') }}` in build and rename build to host?

File: recipe/meta.yaml (hunk 4 above)
[the remark this answers is not in the document]
Comment (sodre) [reply recipe/meta.yaml]: sorry to ask.. how can I pass two conda_build_config.yaml files?

[the remark this answers is not in the document]
Comment (isuruf) [reply recipe/meta.yaml]: `-m file1 -m file2`.

[the remark this answers is not in the document]
Comment (sodre) [reply recipe/meta.yaml]: okay, in that case the "json-c" bug came back. Look at the run requirements lines.

```yaml
[conda@linux-anvil:~/git/postgis-feedstock/recipe] [devtoolset-2] [base] build-variants
± conda render -m conda_build_config.yaml -m /opt/conda/conda_build_config.yaml .
package:
    name: postgis
    version: 2.4.3
source:
    fn: postgis-2.4.3.tar.gz
    patches:
        - 0001-gdal-test.patch
    sha256: ea5374c5db6b645ba5628ddcb08f71d3b3d90a464d366b4e1d20d5a268bde4b9
    url: http://download.osgeo.org/postgis/source/postgis-2.4.3.tar.gz
build:
    number: '0'
requirements:
    build:
        - bzip2 1.0.6 1
        - ca-certificates 2018.4.16 0
        - cunit 2.1.3 0
        - expat 2.2.5 0
        - geos 3.6.2 1
        - gettext 0.19.8.1 0
        - giflib 5.1.4 0
        - icu 58.2 0
        - intel-openmp 2018.0.0 8
        - jpeg 9b 2
        - json-c 0.12.1 0
        - krb5 1.14.6 0
        - libffi 3.2.1 3
        - libgcc-ng 7.2.0 hdf63c60_3
        - libgfortran 3.0.0 1
        - libgfortran-ng 7.2.0 hdf63c60_3
        - libiconv 1.15 0
        - libuuid 1.0.3 1
        - ncurses 5.9 10
        - pcre 8.41 1
        - pixman 0.34.0 2
        - poppler-data 0.4.9 0
        - proj4 4.9.3 5
        - tk 8.6.7 0
        - toolchain 2.1.3 0
        - util-linux 2.21 0
        - xz 5.2.3 0
        - zlib 1.2.11 0
        - boost-cpp 1.66.0 1
        - freexl 1.0.5 0
        - glib 2.55.0 0
        - hdf4 4.2.13 0
        - hdf5 1.10.1 2
        - libpng 1.6.34 0
        - libtiff 4.0.9 0
        - libxml2 2.9.8 0
        - mkl 2018.0.2 1
        - openssl 1.0.2o 0
        - perl 5.26.0 hae598fd_0
        - readline 7.0 0
        - xerces-c 3.2.0 0
        - freetype 2.8.1 0
        - geotiff 1.4.2 1
        - kealib 1.4.7 4
        - libpq 9.6.3 0
        - libssh2 1.8.0 2
        - libxslt 1.1.32 0
        - openjpeg 2.3.0 2
        - sqlite 3.20.1 2
        - curl 7.59.0 1
        - fontconfig 2.12.6 0
        - libspatialite 4.3.0a 19
        - python 3.6.5 1
        - cairo 1.14.10 0
        - certifi 2018.4.16 py36_0
        - libdap4 3.18.3 2
        - libnetcdf 4.6.1 2
        - numpy-base 1.14.3 py36hdbf6ddf_0
        - postgresql 9.6.8 py36_0
        - poppler 0.61.1 3
        - setuptools 39.1.0 py36_0
        - wheel 0.31.0 py36_0
        - pip 9.0.3 py36_0
        - boost 1.66.0 py36_1
        - libkml 1.3.0 6
        - libgdal 2.2.4 2
        - gdal 2.2.4 py36_0
        - mkl_fft 1.0.2 py36_0
        - mkl_random 1.0.1 py36_0
        - numpy 1.14.3 py36h14a74c5_0
    run:
        - pcre
        - proj4 >=4.9.3,<4.9.4.0a0
        - gdal >=2.2.4,<2.3.0a0
        - geos >=3.6.2,<3.6.3.0a0
        - postgresql >=9.6.8,<9.7.0a0
        - json-c
        - gettext
test:
    commands:
        - shp2pgsql
        - pgsql2shp
        - raster2pgsql
about:
    description: 'PostGIS is a spatial database extender for PostgreSQL object-relational

        database. It adds support for geographic objects allowing location

        queries to be run in SQL.

        '
    dev_url: http://svn.osgeo.org/postgis/
    doc_url: http://postgis.net/docs/manual-2.3/
    home: http://postgis.net
    license: GPL2
    license_family: GPL
    license_file: LICENSE.TXT
    summary: PostGIS adds geometry, geography, raster and other types to the PostgreSQL
        database.
extra:
    copy_test_source_files: true
    final: true
    recipe-maintainers:
        - sodre
        - dwnoble

package:
    name: postgis
    version: 2.4.3
source:
    fn: postgis-2.4.3.tar.gz
    patches:
        - 0001-gdal-test.patch
    sha256: ea5374c5db6b645ba5628ddcb08f71d3b3d90a464d366b4e1d20d5a268bde4b9
    url: http://download.osgeo.org/postgis/source/postgis-2.4.3.tar.gz
build:
    number: '0'
requirements:
    build:
        - bzip2 1.0.6 1
        - ca-certificates 2018.4.16 0
        - cunit 2.1.3 0
        - expat 2.2.5 0
        - geos 3.6.2 1
        - gettext 0.19.8.1 0
        - giflib 5.1.4 0
        - icu 58.2 0
        - intel-openmp 2018.0.0 8
        - jpeg 9b 2
        - json-c 0.12.1 0
        - krb5 1.14.6 0
        - libffi 3.2.1 3
        - libgcc-ng 7.2.0 hdf63c60_3
        - libgfortran 3.0.0 1
        - libgfortran-ng 7.2.0 hdf63c60_3
        - libiconv 1.15 0
        - libuuid 1.0.3 1
        - ncurses 5.9 10
        - pcre 8.41 1
        - pixman 0.34.0 2
        - poppler-data 0.4.9 0
        - proj4 4.9.3 5
        - tk 8.6.7 0
        - toolchain 2.1.3 0
        - util-linux 2.21 0
        - xz 5.2.3 0
        - zlib 1.2.11 0
        - boost-cpp 1.66.0 1
        - freexl 1.0.5 0
        - glib 2.55.0 0
        - hdf4 4.2.13 0
        - hdf5 1.10.1 2
        - libpng 1.6.34 0
        - libtiff 4.0.9 0
        - libxml2 2.9.8 0
        - mkl 2018.0.2 1
        - openssl 1.0.2o 0
        - perl 5.26.0 hae598fd_0
        - readline 7.0 0
        - xerces-c 3.2.0 0
        - freetype 2.8.1 0
        - geotiff 1.4.2 1
        - kealib 1.4.7 4
        - libpq 9.6.3 0
        - libssh2 1.8.0 2
        - libxslt 1.1.32 0
        - openjpeg 2.3.0 2
        - sqlite 3.20.1 2
        - curl 7.59.0 1
        - fontconfig 2.12.6 0
        - libspatialite 4.3.0a 19
        - python 3.6.5 1
        - cairo 1.14.10 0
        - certifi 2018.4.16 py36_0
        - libdap4 3.18.3 2
        - libnetcdf 4.6.1 2
        - numpy-base 1.14.3 py36hdbf6ddf_0
        - postgresql 10.3 py36_0
        - poppler 0.61.1 3
        - setuptools 39.1.0 py36_0
        - wheel 0.31.0 py36_0
        - pip 9.0.3 py36_0
        - boost 1.66.0 py36_1
        - libkml 1.3.0 6
        - libgdal 2.2.4 2
        - gdal 2.2.4 py36_0
        - mkl_fft 1.0.2 py36_0
        - mkl_random 1.0.1 py36_0
        - numpy 1.14.3 py36h14a74c5_0
    run:
        - pcre
        - proj4 >=4.9.3,<4.9.4.0a0
        - gdal >=2.2.4,<2.3.0a0
        - geos >=3.6.2,<3.6.3.0a0
        - postgresql >=10.3,<10.4.0a0
        - json-c
        - gettext
test:
    commands:
        - shp2pgsql
        - pgsql2shp
        - raster2pgsql
about:
    description: 'PostGIS is a spatial database extender for PostgreSQL object-relational

        database. It adds support for geographic objects allowing location

        queries to be run in SQL.

        '
    dev_url: http://svn.osgeo.org/postgis/
    doc_url: http://postgis.net/docs/manual-2.3/
    home: http://postgis.net
    license: GPL2
    license_family: GPL
    license_file: LICENSE.TXT
    summary: PostGIS adds geometry, geography, raster and other types to the PostgreSQL
        database.
extra:
    copy_test_source_files: true
    final: true
    recipe-maintainers:
        - sodre
        - dwnoble
```

**Comment by isuruf** [comment]
@conda-forge-admin, please rerender

**Pull request merged**